websites/serializers.py

```diff
@@ -37,14 +39,25 @@ class Meta:
             "publish_date",
             "metadata",
             "starter",
+            "owner",
         ]
+        extra_kwargs = {"owner": {"write_only": True}}
 
 
 class WebsiteDetailSerializer(serializers.ModelSerializer):
     """ Serializer for websites with serialized config """
 
     starter = WebsiteStarterDetailSerializer(read_only=True)
 
+    def create(self, validated_data):
+        """Ensure that the website is created by the requesting user"""
+        request = self.context.get("request")
+        if request and hasattr(request, "user") and isinstance(request.user, User):
+            validated_data["owner"] = request.user
+            with transaction.atomic():
+                website = super().create(validated_data)
+            return website
```

Comment:
Wouldn't it make more sense for the view to be responsible for adding the `owner` to the request? This just seems to make the serializer less flexible. In general I think the more a serializer knows about the request, the less usable it is.

Reply:
This is replicating the same strategy used in open-discussions [here](https://github.com/mitodl/open-discussions/blob/master/course_catalog/serializers.py#L590-L593), but I can change it.

Reply:
Change made